### DISCUSSION TITLE
Spinners can show a huge negative SPM number

### DISCUSSION THREAD

**Discussion by Gabixel** [ppy/osu#21210] [Q&A] [answered]
I initially saw this long succession of zeros in the editor, but it was so random that I couldn't get a screenshot and reproduce it.
It might refer to https://github.com/ppy/osu/issues/10455, but since it also happens with replays I guess it has to be handled differently.

Today, randomly looking at one of my replays, I spammed the right arrow at the end of the map timeline (where there was a spinner), and it would seem to give the same result that I saw in the editor a while ago (wait for the gif):

![spinner](https://user-images.githubusercontent.com/43073074/201474068-e4998e7a-77ab-4615-b572-d3fcb892b53c.gif)

By moving the timeline back slightly you can also pause without being transported to the result screen. Consequently, you can inspect the large number:

![image](https://user-images.githubusercontent.com/43073074/201474217-d0b06c89-45d2-48e4-856e-301f48f56e88.png)

I am not a great expert here, but checking in the C# documentation, this number seems to look a lot like the minimum value of a `double`, at least for the initial digits:

![image](https://user-images.githubusercontent.com/43073074/201474280-a15141ef-776b-43a9-9358-a734cef0632f.png)

I couldn't reproduce it with `AutoPlay`, even after trying the various mod combinations I used in my replay. So it may be a case-specific to how I rotated the spinner.

In case you want to test and reproduce,
- [here's the replay (as usual, rename it to `.osr`)](https://github.com/ppy/osu/files/9994976/Gabixel.playing.S3RL.feat.Tamika.-.Rainbow.Girl.Candlelight.mix.Anto.Hard.zip)
- [here's the map](https://osu.ppy.sh/beatmapsets/1074368#osu/2248333)

I can imagine that the easiest solution is to limit the minimum value to 0. I doubt that anyone can get negative spins per minute

(Version is `2022.1101.0-lazer`)

**Comment by smoogipoo** [comment, marked as answer]
https://github.com/ppy/osu/issues/21221
